absl/synchronization/internal/create_thread_identity.cc

```diff
@@ -67,6 +67,33 @@ static intptr_t RoundUp(intptr_t addr, intptr_t align) {
   return (addr + align - 1) & ~(align - 1);
 }
 
+namespace {
+
+  void ResetThreadIdetity(base_internal::ThreadIdentity* identity) {
+    base_internal::PerThreadSynch& pts = identity->per_thread_synch;
+    pts.next = nullptr;
+    pts.skip = nullptr;
+    pts.may_skip = false;
+    pts.waitp = nullptr;
+    pts.suppress_fatal_errors = false;
+    pts.readers = 0;
+    pts.priority = 0;
+    pts.next_priority_read_cycles = 0;
+    pts.state = base_internal::PerThreadSynch::State::kAvailable;
+    pts.maybe_unlocking = false;
+    pts.wake = false;
+    pts.cond_waiter = false;
+    pts.all_locks = nullptr;
+    identity->waiter_state = {};
+    identity->blocked_count_ptr = nullptr;
```

Comment:
Please use relaxed stores (`store(nullptr, std::memory_order_relaxed)`) for this and the other atomics.

Reply:
`blocked_count_ptr` is a pointer to an atomic int. Is it okay to just reset the pointer or do I first need to store a `0` into the atomic int?